### ISSUE TITLE
"torch.cat Error in StableZero123_Conditioning: Expected all tensors to be on the same device"

### ISSUE THREAD

**Issue by PILIPINGMI0** [comfyanonymous/ComfyUI#2646]
### Issue Description  01
While using the StableZero123_Conditioning in ComfyUI, I encountered an error related to the `torch.cat` operation. The error message indicates **"Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"**

Error occurred when executing StableZero123_Conditioning:

Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument tensors in method wrapper_CUDA_cat)

File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 154, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 84, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 77, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_stable3d.py", line 49, in encode
cond = torch.cat([pooled, cam_embeds.repeat((pooled.shape[0], 1, 1))], dim=-1)

           I'm newbie but tried to fix it with chatgpt's help and it worked: I changed code line49 / line 91  to
           ***>>> cond = torch.cat([pooled.to("cuda:0"), cam_embeds.repeat((pooled.shape[0], 1, 1)).to("cuda:0")], dim=-1)***, 
           but it will be replaced everytime I git pull.

### Issue Description  02
Simillar Issue appears:
While executing the ImageBlend plugin in ComfyUI, I encountered anther error related to the `torch.cat` operation. The error message indicates **"Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!"**

Error occurred when executing ImageBlend:

Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu!

File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 155, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 85, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 78, in map_node_over_list
results.append(getattr(obj, func)(**slice_dict(input_data_all, i)))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\comfy_extras\nodes_post_processing.py", line 42, in blend_images
blended_image = image1 * (1 - blend_factor) + blended_image * blend_factor

### Issue Description  03
this issue is related to **ComfyUI-Impact-Pack**  

Error occurred when executing ImageListToImageBatch:

Expected all tensors to be on the same device, but found at least two devices, cuda:0 and cpu! (when checking argument for argument tensors in method wrapper_CUDA_cat)

File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 155, in recursive_execute
output_data, output_ui = get_output_data(obj, input_data_all)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 85, in get_output_data
return_values = map_node_over_list(obj, input_data_all, obj.FUNCTION, allow_interrupt=True)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\execution.py", line 69, in map_node_over_list
results.append(getattr(obj, func)(**input_data_all))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "D:\1 APPS\ComfyUI_windows_portable\ComfyUI\custom_nodes\ComfyUI-Impact-Pack\modules\impact\util_nodes.py", line 327, in doit
image1 = torch.cat((image1, image2), dim=0)

### Environment Information
- Operating System: Windows 11
- Python Version: 3.11.6
- PyTorch Version: 1.2.1+cu121

Encountering a recurring error across multiple plugins in ComfyUI: "Expected all tensors to be on the same device, but found at least two devices, cuda:0."  Seeking an resolution approach to ensure consistent device usage during torch.cat operations.

**Comment by comfyanonymous** [comment]
Does this fix it? https://github.com/comfyanonymous/ComfyUI/commit/7f4725f6b3f72dd8bdb60dae5dd2c3e943263bcf

**Comment by PILIPINGMI0** [comment]
> Does this fix it? [7f4725f](https://github.com/comfyanonymous/ComfyUI/commit/7f4725f6b3f72dd8bdb60dae5dd2c3e943263bcf)

YES!! I replaced two raw files and the workflow now works fine!!! thx for help!  xx
